### PR TITLE
memory leak fixed

### PR DESCRIPTION
**Pull request by AFranci**
The auxiliar structures of Triangle and Tetgen are now properly erased.
Another memory leak source in BuildCompositeConditions has been fixed.

### CODE REVIEW

**Comment by maceligueta** [comment]
Can we merge now to master? @pooyan-dadvand said something about not merging to master. 
In my opinion, this is a very important bug and should be merged to both branches (master and release5.1).

**Comment by AFranci** [comment]
I agree

File: applications/PfemBaseApplication/custom_utilities/mesh_data_transfer_utilities.cpp [hunk 1]
```diff
@@ -1382,6 +1382,7 @@ namespace Kratos
 		//ELEMENT TRANSFER CHECK//
 	      }
 
+	    temporal_elements.clear();
```
Comment (josep-m-carbonell) [inline applications/PfemBaseApplication/custom_utilities/mesh_data_transfer_utilities.cpp:1385]: Is it needed anymore?  temporal_elements are destroyed at the end of the method.

**Review by josep-m-carbonell — APPROVED** [review]
Thank you Alessandro to locate the memory leaks

File: applications/PfemBaseApplication/custom_utilities/mesh_data_transfer_utilities.cpp (hunk 1 above)
Comment (AFranci) [reply applications/PfemBaseApplication/custom_utilities/mesh_data_transfer_utilities.cpp:1385]: I will check it!

**Comment by pooyan-dadvand** [comment]
It was before creating the release branch. Now everybody can continue his normal work on master branch. By the way for such a important bug fix, it would be great to merge it also in release branch.

**Pull request merged**